explore/src/main/scala/explore/targeteditor/AladinCell.scala

```diff
@@ -66,7 +71,28 @@ object AladinCell extends ModelOptics {
       // flag to trigger centering. This is a bit brute force but
       // avoids us needing a ref to a Fn component
       .useStateViewWithReuse(false)
-      .useEffectWithDepsBy((p, _, _, _) => (p.uid, p.tid)) { (props, _, options, _) => _ =>
+      // GuideStar candidates
+      .useState(List.empty[GuideStarCandidate])
+      // Lisen on web worker for messages with catalog candidates
+      .useEffectOnMountBy((props, _, _, _, gs) =>
+        props.ctx.worker.stream
+          .evalMap(m =>
+            decodeFromTransferable[IO, CatalogResults](m)(gsl => gs.setState(gsl.candidates).to[IO])
+          )
+          .compile
+          .drain
+          .whenA(props.obsConf.isDefined)
+      )
```

Comment:
Can we use a `StreamRenderer` (from crystal) ?

`StreamRenderer` does exactly this: reads from a stream and sets a state whenever a new value is read.

Reply:
It will also cancel the fiber on unmount.

Reply:
Ah, nice tip

Reply:
Ah but Stream renderer creates a component right? whil the code above only maniulates the state

Reply:
Right, you can save the component in state.

Or, on second thought, we should have a `useStream` hook. I'll check that approach.